ocaml-lsp-server/src/custom_requests/req_merlin_call_compatible.ml

```diff
@@ -0,0 +1,156 @@
+open Import
+
+let capability = ("handleMerlinCallCompatible", `Bool true)
+
+let meth = "ocamllsp/merlinCallCompatible"
+
+type params =
```

Comment:
This type should be in its own submodule and should have the following two functions:

```ocaml
val yojson_of_t : t -> Json.t
val t_of_yojson : Json.t -> t
```

Even if the two functions above are manually written, they should conform to our API.

Reply:
Regarding https://github.com/ocaml/ocaml-lsp/pull/1265#pullrequestreview-2070665576, I'll probably go for derivation. But may I ask why (it is about curiosity)? Since in other custom query, This is sometimes the case (in a `Request_param` module, which is never exposed) sometimes not (ie: when a list of `uri` is requested), so does this have a different implication (code-style, additional metrics?)

Reply:
Looks good to me. The reason why we always use the interface above is to make it trivial to switch between manually written json converters and auto generated ones.